src/main/java/backend/model/payment/SponsoringInvoice.kt

```diff
@@ -156,9 +156,14 @@ class SponsoringInvoice : Invoice {
                 return total
             }
             is Sponsor -> listOf(EmailAddress(registeredSponsor!!.email))
-            else -> throw Exception("No sponsor email found for invoice $id")
+            else -> listOf()
         }
     }
+
+    fun removeSponsor() {
```

Comment:
sounds like "delete" maybe we should call it "unassignSponsor"

Reply:
same for the other "removeSponsor" methods, I did notice as the naming somehow implied for me that the whole sponsor is removed, not only the userdata.

Reply:
Cool I'll rename them